agent/consul/state/catalog.go

```diff
@@ -3450,6 +3450,10 @@ func parseNodes(tx ReadTxn, ws memdb.WatchSet, idx uint64,
 		ws.AddWithLimit(watchLimit, services.WatchCh(), allServicesCh)
 		for service := services.Next(); service != nil; service = services.Next() {
 			ns := service.(*structs.ServiceNode).ToNodeService()
+			//Update Node Meta with version from Consul Service - for backward compatibility
+			if _, ok := dump.Meta[structs.MetaConsulVersion]; !ok && ns.ID == "consul" && ns.Meta["version"] != "" {
+				dump.Meta[structs.MetaConsulVersion] = ns.Meta["version"]
+			}
```

Comment:
can we consider including the change in our unit test case `TestStateStore_NodeInfo_NodeDump` in `catalog_test.go`?

Reply:
@huikang , Please note - this change is not there now. The above approach would work only if agent is running as server & Leader (due to RPC Forward).
So, now a better robust approach is taken to get consul version irrespective of agent running as client or server and non-leader/leader. The approach is to get all serf members and get version from 'build' tag. Hence above specific code change is discarded and therefore the testcase.

Reply:
@huikang , this was added as fallback, if for some reason not able to get version from serf members.
Updated the test case in TestStateStore_NodeInfo_NodeDump.
<img width="1299" alt="Screenshot 2023-07-01 at 7 05 13 PM" src="https://github.com/hashicorp/consul/assets/3139634/3eff1a22-1594-459f-bf5a-812523a3458b">